analyzers/tests/SonarAnalyzer.Test/TestCases/DeadStores.RoslynCfg.cs

```diff
@@ -1515,4 +1515,51 @@ public void TryReturns_Loop()
 
         private void SomethingThatCanThrow() { }
     }
+
+    public class PeachValidation
+    {
+        // https://github.com/SonarSource/sonar-dotnet/issues/9471
+        void Repro9471(bool condition)
+        {
+            byte[]? tempArray = null;
+            byte[] anArray = condition ? tempArray = new byte[42] : null; // Noncompliant FP
+            if (tempArray != null)
+            {
+                Console.WriteLine(tempArray);
+            }
+        }
+
+        // https://github.com/SonarSource/sonar-dotnet/issues/9472
+        void Repro9472()
+        {
+            char ch;
+            switch (ch = GetAChar())   // Noncompliant FP
+            {
+                case 'A':
+                    break;
+                case 'B':
+                    Console.WriteLine(ch);
+                    break;
+                default:
+                    Console.WriteLine("Something");
+                    break;
+            }
+            char GetAChar() => 'c';
+        }
+
+        // https://github.com/SonarSource/sonar-dotnet/issues/9473
+        void Repro9473(IDisposable data)
+        {
+            using (data = Something()) // Noncompliant
+            {
+                data = Something();    // Noncompliant FP
+            }
+
+            if (data != null)
+            {
+                Console.WriteLine("Hello");
+            }
```

Comment:
Is the `if` needed for minimal reproducer, or can `Console.WriteLine(data) also does the trick?